var/spack/repos/builtin/packages/r-bookdown/package.py

```diff
@@ -10,13 +10,16 @@ class RBookdown(RPackage):
     """Output formats and utilities for authoring books and technical
     documents with R Markdown."""
 
-    homepage = "https://cran.r-project.org/package=bookdown"
-    url      = "https://cran.rstudio.com/src/contrib/bookdown_0.5.tar.gz"
-    list_url = "https://cran.rstudio.com/src/contrib/Archive/bookdown"
+    homepage = "https://cloud.r-project.org/package=bookdown"
+    url      = "https://cloud.r-project.org/src/contrib/bookdown_0.5.tar.gz"
+    list_url = "https://cloud.r-project.org/src/contrib/Archive/bookdown"
 
+    version('0.12', sha256='38eb4c5b877ccd85b16cfe74a48c3bc53de2f276da98e5515f37e7a06e065bb0')
     version('0.5', '7bad360948e2b22d28397870b9319f17')
 
-    depends_on('r-yaml@2.1.14:', type=('build', 'run'))
-    depends_on('r-rmarkdown@1.5:', type=('build', 'run'))
-    depends_on('r-knitr@1.16:', type=('build', 'run'))
+    depends_on('r-yaml@2.1.14:', when='@:0.10', type=('build', 'run'))
+    depends_on('r-rmarkdown@1.12:', type=('build', 'run'))
+    depends_on('r-knitr@1.22:', type=('build', 'run'))
     depends_on('r-htmltools@0.3.6:', type=('build', 'run'))
+    depends_on('r-xfun@0.6:', when='@0.12:', type=('build', 'run'))
+    depends_on('r-tinytex@0.12:', when='@0.12:', type=('build', 'run'))
```

Comment:
One more requirement:

> SystemRequirements:	Pandoc (>= 1.17.2)

Reply:
This is actually kind of a big one. Pandoc is a Haskell package, which has a really odd build system, ultimately requiring a binary version of GHC to build GHC. There are binaries of pandoc available for MacOS and Linux and I wonder if that would be a better way to go. In any event, I will put his on my list of things to work on, but since the pandoc dependency has been missing from the beginning, and it will take a while to sort out how to best get that into spack, I do not think this should hold up this PR.

Reply:
Indeed, the issue with GHC/pandoc was discussed extensively in #1408.

Reply:
At the very least, you should add a commented out dependency and a TODO note.

Reply:
That might be best for now. After reading through #1408 I am inclined to go with a binary install of pandoc. Given that this dependency was never in place I wonder if this package actually works, at least in the case where pandoc is not installed as a system package. Is there a way to mark a package broken in spack? I suppose something could be done with a `conflicts` statement.

Reply:
@adamjstewart I am looking into creating a pandoc package with the binaries that are being produced from version 2.0+. I would like for this to work on Linux and Mac but I am not sure how to handle the checksums in `version()`. There could be two downloads, depending on the platform:

1. pandoc-2.7.3-linux.tar.gz
1. pandoc-2.7.3-macOS.zip

I know there would be a `url_for_version` needed but how do I handle the two different checksums in `version()` for the different downloads?

Reply:
There's currently no way to do this. The best you can do is something like:
```python
import platform

class Foo(FooPackage):
    if platform.system() == 'darwin':
        # macOS versions
        version('1.2.3', sha256='...')
    elif platform.system() == 'linux':
        # Linux versions
        version('1.2.3', sha256='...')
```

Reply:
@adamjstewart I added a pandoc package.

Reply:
Added dependency on pandoc to r-bookdown.